src/go/k8s/internal/util/client.go

```diff
@@ -0,0 +1,353 @@
+package util
```

Comment:
I'm not a huge fan of "kitchen sink"/"junk draw" packages. They create dependency and discoverability headaches.

One pattern I've seen that felt like a good middleground was an organized... kitchen sink.

```
pkg/
  util/
    kubeutil/   # Utilities for Kubernetes
    ptrutil/      # Utilities for prt types
    <domain>util/
```